core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -174,6 +177,24 @@ private[spark] class CoarseGrainedExecutorBackend(
 
 private[spark] object CoarseGrainedExecutorBackend extends Logging {
 
+  private def addMesosDelegationTokens(driverConf: SparkConf) {
+    val value = driverConf.get("spark.mesos.kerberos.userCredentials")
+    val tokens = DatatypeConverter.parseBase64Binary(value)
+
+    logDebug(s"Found delegation tokens of ${tokens.length} bytes.")
+
+    // Use tokens for HDFS login.
+    val hadoopConf = SparkHadoopUtil.get.newConfiguration(driverConf)
+    hadoopConf.set("hadoop.security.authentication", "Token")
+    UserGroupInformation.setConfiguration(hadoopConf)
+
+    // Decode tokens and add them to the current user's credentials.
+    val creds = UserGroupInformation.getCurrentUser.getCredentials
+    val tokensBuf = new java.io.ByteArrayInputStream(tokens)
+    creds.readTokenStorageStream(new java.io.DataInputStream(tokensBuf))
+    UserGroupInformation.getCurrentUser.addCredentials(creds)
+  }
```

Comment:
Yes, I believe @jerryshao is working on a standard method of delegating tokens.  We can plug that in once it exists.